islandora_large_image.install

```diff
@@ -9,33 +9,92 @@
-      $command = 'kdu_compress >NUL 2>&1';
+  if (in_array($phase, array('install', 'runtime'))) {
+    $has_kakadu = _islandora_large_image_install_has_kakadu();
+    $has_jp2_imagemagick = _islandora_large_image_install_has_imagemagick_with_jpeg2000_support();
+    if ($has_kakadu) {
+      $requirements['islandora_large_image_kakadu'] = array(
+        'title' => $t("Islandora Solution Pack Large Image: Kakadu Image Compression"),
+        'value' => $t("Installed"),
+        'severity' => REQUIREMENT_INFO,
+        'description' => $t('The kdu_compress executable is accessible.'),
+      );
     }
-
-    exec($command, $out, $ret);
-
-    if ($ret != 0) {
-      $requirements['kakadu'] = array(
-        'title' => $t("Kakadu Image Compression"),
+    else {
+      $requirements['islandora_large_image_kakadu'] = array(
+        'title' => $t("Islandora Solution Pack Large Image: Kakadu Image Compression"),
+        'value' => $t("Not installed"),
+        'severity' => REQUIREMENT_INFO,
+        'description' => $t('The kdu_compress executable is not accessible.'),
+      );
+    }
+    if ($has_jp2_imagemagick) {
+      $requirements['islandora_large_image_imagemagick'] = array(
+        'title' => $t("Islandora Solution Pack Large Image: Imagemagick Image Compression"),
+        'value' => $t("Installed"),
+        'severity' => REQUIREMENT_INFO,
+        'description' => $t('Imagemagick is present and appears to support JPEG2000.'),
+      );
+    }
+    else {
+      $requirements['islandora_large_image_imagemagick'] = array(
+        'title' => $t("Islandora Solution Pack Large Image: Imagemagick Image Compression"),
+        'value' => $t("Not installed"),
+        'severity' => REQUIREMENT_INFO,
+        'description' => $t('Imagemagick does not appears to support JPEG2000.'),
+      );
+    }
+    if (!($has_kakadu || $has_jp2_imagemagick)) {
+      $requirements['islandora_large_image_encoder'] = array(
+        'title' => $t("Islandora Solution Pack Large Image: JPEG 2000 image encoder"),
         'value' => $t("Not installed"),
         'severity' => REQUIREMENT_ERROR,
-        'description' => $t('Ensure that adore-djatoka is installed and that kdu_compress is accessible by PHP.'),
+        'description' => $t('A JPEG 2000 encoder is not installed/configured.'),
+      );
+    }
+    else {
+      $requirements['islandora_large_image_encoder'] = array(
+        'title' => $t("Islandora Solution Pack Large Image: JPEG 2000 image encoder"),
+        'value' => $t("Installed"),
+        'severity' => REQUIREMENT_OK,
+        'description' => $t('A JPEG 2000 encoder is installed.'),
       );
     }
   }
   return $requirements;
 }
 
+/**
+ * Helper to test if Kakadu appears to be available.
+ */
+function _islandora_large_image_install_has_kakadu() {
+  module_load_include('inc', 'islandora', 'includes/utilities');
+
+  $out = array();
+  $ret = 1;
+
+  $command = 'command -v kdu_compress >/dev/null 2>&1';
```

Comment:
Sorry for the delay, out of office time, release duties and personal issues. I have some doubts about how the checking if required Kakadu binary is present happens during install and runtime here.. this command here assumes that `kdu_compress` is globally available and not only present in some secure /something/bin location. I would assume that during the runtime phase of this invocation, 'islandora_kakadu_url' is already set or at least to be trusted, which means, we should use variable_get(islandora_kakadu_url', '/usr/bin/kdu_compress'); and during install we should try with '/usr/bin/kdu_compress' instead which is our default right?.

Reply:
Unknown... I didn't really look too deeply into the specifics myself, but just moved it around... as in, it is the same as [the old implementation](https://github.com/Islandora/islandora_solution_pack_large_image/pull/182/files#diff-88c4733c33e0d05cf67c4a3264e8af8cL19).

The [README appears to attempt to describe symlinking `kdu_compress` in the `PATH`](https://github.com/Islandora/islandora_solution_pack_large_image/pull/182/files#diff-04c6e90faac2675aa89e2176d2eec7d8R32)... addressing this inconsistency sounds like another ticket?

Reply:
Hi, I agree but sorry, we are running thinner and thinner on commits lately, so would hate to leave that one untouched and forgotten. Would it be too much to ask you to please just add, either that variable get for run time and and the default itself for install (which really is what the derivative process uses once deployed, right?) to this pull? Maybe I'm asking too much. Sorry, still on release momentum. @ajstanley what do you think?

Reply:
@DiegoPino For the sake of expediency we can merge what we have (it works) and address the pathing issue from Adam's new ticket.

I should have some time this afternoon to work on that one.  I'll assign it to myself.